### ISSUE TITLE
Invalid go code: Missing data types

### ISSUE THREAD

**Issue by andygrunwald** [a-h/generate#15]
If you compile this:

```json
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "name": "Repository Configuration",
    "type": "object",
    "additionalProperties": false,
    "required": [ "name" ],
    "properties": {
        "name": {
            "type": "string",
            "description": "Repository name."
        },
        "abandoned": {
            "type": "object",
            "description": "List of packages marked as abandoned for this repository, the mark can be boolean or a package name/URL pointing to a recommended alternative.",
            "additionalProperties": {
                "type": ["boolean", "string"],
                "description": "A valid Package name"
            }
        }
    }
}
```

you will get this as a result:

```go
package main

type Root struct {
  Abandoned * `json:"abandoned,omitempty"`
  Name string `json:"name,omitempty"`
}
```

As you see the var `Abandoned` as a pointer reference, but the type is missing.

**Comment by maciekmm** [comment]
I don't think whether this should be considered a bug. You can specify _title_ property in "abandoned section to have the struct generated. It'd be a cool feature to have the name derived from the parent property tho.

**Comment by a-h** [comment]
So the fix above will generate the Abandoned struct, even though it doesn't have properties resulting in valid code.

Also, the go code that's generated is safe to use to unmarshal JSON into the struct (although it won't be maintained if the go struct is marshalled back into JSON).

Is that a good outcome for this scenario? Or can you think of a better one?

**Comment by a-h** [comment]
I'm going to close this on given I've merged #18. Thanks!
